resource-sharing-partners/src/main/java/no/sikt/rsp/ContactInfoConverter.java

```diff
@@ -0,0 +1,164 @@
+package no.sikt.rsp;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.Optional;
+import no.nb.basebibliotek.generated.Record;
+import no.sikt.alma.generated.Address;
+import no.sikt.alma.generated.Address.AddressTypes;
+import no.sikt.alma.generated.Address.Country;
+import no.sikt.alma.generated.Addresses;
+import no.sikt.alma.generated.ContactInfo;
+import no.sikt.alma.generated.Email;
+import no.sikt.alma.generated.Email.EmailTypes;
+import no.sikt.alma.generated.Emails;
+import no.sikt.alma.generated.Phone;
+import no.sikt.alma.generated.Phone.PhoneTypes;
+import no.sikt.alma.generated.Phones;
+import nva.commons.core.JacocoGenerated;
+import nva.commons.core.StringUtils;
+
+public class ContactInfoConverter {
+
+    private static final boolean BEST_EMAIL_IS_ALWAYS_PREFERRED = true;
+    private static final boolean THERE_IS_ONLY_ONE_PHONE_SO_IT_IS_ALWAYS_PREFERRED = true;
+    private static final boolean P_ADDRESS_IS_ALWAYS_PREFERRED = true;
+    private static final List<String> PHONE_TYPES = List.of("claimPhone", "orderPhone", "paymentPhone", "returnsPhone");
+    private static final List<String> ADDRESS_TYPES = List.of("billing", "claim", "order", "payment", "returns",
+                                                              "shipping");
+    private static final List<String> EMAIL_TYPES = List.of("claimMail", "orderMail", "paymentMail", "queries",
+                                                            "returnsMail");
+
+    @JacocoGenerated
+    public ContactInfoConverter() {
+    }
+
+    public static ContactInfo extractContactInfoFromRecord(Record record) {
+        var contactInfo = new ContactInfo();
+        contactInfo.setAddresses(extractAddressesFromRecord(record));
+        contactInfo.setPhones(createPhones(record));
+        contactInfo.setEmails(createEmails(record));
+        return contactInfo;
+    }
+
+    private static Emails createEmails(Record record) {
+        var emailBest = createEmailBest(record);
+        var emailBestExists = emailBest.isPresent();
+        var emailRegular = createEmailRecgular(record, emailBestExists);
+        var emails = new Emails();
+        emailBest.ifPresent(email -> emails.getEmail().add(email));
+        emailRegular.ifPresent(email -> emails.getEmail().add(email));
+        return emails;
+    }
+
+    private static Optional<Email> createEmailRecgular(Record record, boolean emailBestExists) {
```

Comment:
Skal det være "Regular" isteden for "Recgular"?

Reply:
fikset